src/include/mpir_request.h

```diff
@@ -149,6 +149,11 @@ struct MPIR_Request {
      * 32 bytes and 32-bit integers */
     MPIR_cc_t cc;
 
+#if MPICH_THREAD_USE_MDTA == 1
```

Comment:
This commit should be merged with the one that adds the config option. Also, there needs to be a more descriptive commit message explaining the concept.